infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -229,7 +258,13 @@ public LogReplicationEntry receive(LogReplicationEntry message) {
         if (rxState.equals(RxState.LOG_ENTRY_SYNC)) {
             return logEntrySinkBufferManager.processMsgAndBuffer(message);
         } else {
-            return snapshotSinkBufferManager.processMsgAndBuffer(message);
+            LogReplicationEntry ack = snapshotSinkBufferManager.processMsgAndBuffer(message);
+            if (message.getMetadata().getMessageMetadataType() == SNAPSHOT_END) {
+                // Notify end of snapshot sync. This is a blocking call.
+                snapshotSyncPlugin.onSnapshotSyncEnd();
```

Comment:
mmm but the snapshotPlugin is available in this class and not in the snapshotSinkBufferManager.

Reply:
ok, yes since we call it after processMsgAndBuffer(), the apply will have completed.

Reply:
Snapshot_end : means the data has been transferred to the standby site and the standby will start to apply the entries from shadow stream to the real streams.

Reply:
Is this a comment or a response to Pankti's comment? 
Because snapshotSinkBufferManager.processMsgAndBuffer applies the data. So when it comes back we're sure it has been applied.

Reply:
Should check the ack message type.

Reply:
@xiaoqin2012 Would it be enough to change that check for:
``if (ack.getMetadata().getMessageMetadataType() == MessageType.SNAPSHOT_REPLICATED)``
as this is the only ACK sent at the end of all received snapshot sync.

Reply:
mm I'm actually thinking that yeah we probably would need to complement this with keeping the latests started snapshot sync TS in the metadata to cover the case of an ongoing snapshot which is canceled (but still completed out of order)